Assets/HoloToolkit-Tests/Input/Scripts/MicStreamDemo.cs

```diff
@@ -1,85 +1,88 @@
+        /// Can boost volume here as desired. 1 is default.
+        /// <remarks>Can be updated at runtime.</remarks> 
         /// </summary>
         public float InputGain = 1;
 
         /// <summary>
-        /// if keepAllData==false, you'll always get the newest data no matter how long the program hangs for any reason, but will lose some data if the program does hang 
-        /// can only be set on initialization
+        /// if keepAllData==false, you'll always get the newest data no matter how long the program hangs for any reason,
+        /// but will lose some data if the program does hang.
+        /// <remarks>Can only be set on initialization.</remarks>
         /// </summary>
-        public bool KeepAllData = false;
+        public bool KeepAllData;
 
         /// <summary>
-        /// Should the mic stream start automatically when this component is enabled?
+        /// If true, Starts the mic stream automatically when this component is enabled.
         /// </summary>
         public bool AutomaticallyStartStream = true;
 
         /// <summary>
-        /// Do you want to hear what the microphone sounds like by listening to the AudioSource in Unity?
+        /// Plays back the microphone audio source though default audio device.
         /// </summary>
-        public bool ListenToAudioSource = true;
+        public bool PlaybackMicrophoneAudioSource = true;
 
         /// <summary>
-        /// The name of the file to which to save audio (for commands that save to a file)
+        /// The name of the file to which to save audio (for commands that save to a file).
         /// </summary>
         public string SaveFileName = "MicrophoneTest.wav";
 
         /// <summary>
-        /// Records estimation of volume from the microphone to affect other elements of the game object
+        /// Records estimation of volume from the microphone to affect other elements of the game object.
         /// </summary>
-        private float averageAmplitude = 0;
+        private float averageAmplitude;
 
         /// <summary>
-        /// how small can our object be in this demo?
+        /// Minimum size the demo cube can be during runtime.
         /// </summary>
-        private float minSize = .3f;
+        [SerializeField]
+        private float minObjectScale = .3f;
+
+        private bool isRunning;
 
         private void OnAudioFilterRead(float[] buffer, int numChannels)
         {
             // this is where we call into the DLL and let it fill our audio buffer for us
             CheckForErrorOnCall(MicStream.MicGetFrame(buffer, buffer.Length, numChannels));
 
             float sumOfValues = 0;
+
             // figure out the average amplitude from this new data
-            for (int i=0; i<buffer.Length; i++)
+            for (int i = 0; i < buffer.Length; i++)
             {
-                sumOfValues += Mathf.Abs(buffer[i]);
+                if (float.IsNaN(buffer[i]))
```

Comment:
I'm unsure where the NaN is from exactly, but it's what is in the buffer from Unity.